ExampleMod/Common/GlobalItems/WeaponWithGrowingDamage.cs

```diff
@@ -0,0 +1,176 @@
+		}
+		public override void SetDefaults(Item item) {
+			if (item.type == ItemID.Snowball) {
+				experience = 1;
+				UpdateValue(item);
+			}
+		}
+		public override void LoadData(Item item, TagCompound tag) {
+			int xp = tag.Get<int>("experience");//Load experience tag
+			GainExperience(item, xp, true);
+		}
+
+		public override void SaveData(Item item, TagCompound tag) {
+			tag["experience"] = experience;//Save experience tag
+		}
+
+		public override void NetSend(Item item, BinaryWriter writer) {
+			writer.Write(experience);
+		}
+
+		public override void NetReceive(Item item, BinaryReader reader) {
+			int xp = reader.ReadInt32();
+			GainExperience(item, xp, true);
+		}
+		
+		public override void OnHitNPC(Item item, Player player, NPC target, int damage, float knockBack, bool crit) {
+			OnHitNPCGeneral(item, player, target, damage, knockBack, crit);
+		}
+
+		public void OnHitNPCGeneral(Item item, Player player, NPC target, int damage, float knockBack, bool crit, Projectile projectile = null) {
+			//The weapon gains experience when hitting an npc.
+			int xp = damage;
+			if (projectile != null)
+				xp /= 2;
+
+			GainExperience(item, xp);
+		}
+
+		public void GainExperience(Item item, int xp, bool setXp = false) {
+			if (setXp) {
+				experience = xp;
+			}
+			else {
+				experience += xp;
+			}
+
+			UpdateValue(item);
+		}
+
+		private void UpdateValue(Item item, int stackChange = 0) {
+			//The goal of UpdateValue is to give the item bonus value based on it's experience.
+			//With stackable items such as Throwing Knives, the item.value is multiplied by the item.stack.  This is not something that is desired for this example.
+
+			//Example: UpdateValue() is called when a stack is combined.
+			//		The original item.stack was 10, the new item.stack is 100.
+			//		The original experience was 40, the current experience is 40.
+			//		The original extra coins from selling would be 
+			//
+			//	Previous Values:
+			//			
+			//			lastBonusValue: 200
+			//			stack: 10
+			//	New Values:
+			//			
+
+			if (stack == int.MinValue)
+				stack = item.stack + stackChange;
+
+			if (CheckStackZero(item))
+				return;
+
+			int lastBonusValuePerItem = lastBonusValue / stack;//lastBonusValue : 200, stack: 10, lastBonusValuePerItem = 20
+			stack = item.stack + stackChange;//item.stack = 10, stackChange = 90, so the new stack will be 100.  stack = 100
+
+			if (CheckStackZero(item))
+				return;
+
+			int newBonusValue = experience * 5;//experience: 40, newBonusValue = 200
+			int newBonusValuePerItem = newBonusValue / stack;//newBonusValue: 200, stack: 100, newBonusValuePerItem = 2
+			item.value += newBonusValuePerItem - lastBonusValuePerItem;//newBonusValuePerItem: 2, lastBonusValuePerItem, 20, item.value += -18
+			lastBonusValue = newBonusValue;//newBonusValue: 200, lastBonusValue = 200
+		}
+
+		private bool CheckStackZero(Item item) {
+			if (stack <= 0) {
+				item.value = ContentSamples.ItemsByType[item.type].value;
```

Comment:
not a good idea, will overwrite value changes from other mods. Shouldn't be needed.
`UpdateValue` can just be
```
item.value -= lastBonusValuePerItem;

lastBonusValuePerItem = experience * 5 / item.stack;
item.value += lastBonusValuePerItem;
```

Reply:
Updated